code/game/objects/items/RPD.dm

```diff
@@ -481,14 +487,14 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
 /obj/item/pipe_dispenser/proc/activate()
 	playsound(get_turf(src), 'sound/items/deconstruct.ogg', 50, TRUE)
 
-/obj/item/pipe_dispenser/proc/mouse_wheeled(datum/source, mob/user, delta_x, delta_y, params)
+/obj/item/pipe_dispenser/proc/mouse_wheeled(datum/source, delta_x, delta_y, params)
 	SIGNAL_HANDLER
 
 	if(delta_y > 0)
 		piping_layer = min(PIPING_LAYER_MAX, piping_layer + 1)
-	else
+	else if(delta_y < 0)
 		piping_layer = max(PIPING_LAYER_MIN, piping_layer - 1)
-	to_chat(user, "<span class='notice'>You set the layer to [piping_layer].</span>")
+	to_chat(source, "<span class='notice'>You set the layer to [piping_layer].</span>")
```

Comment:
Better, but it still tell you that you've set the piping layer if you side-scroll.